Number_types/include/CGAL/internal/Exact_type_selector.h

```diff
@@ -65,9 +66,15 @@ struct Exact_field_selector
 #ifdef CGAL_USE_GMPXX
 { typedef mpq_class Type; };
 #elif defined(CGAL_USE_GMP)
+# if defined(CGAL_USE_BOOST_MP)
+{ typedef boost::multiprecision::mpq_rational Type; };
+# else
 { typedef Gmpq Type; };
+# endif
```

Comment:
Yes. The goal is exactly the same as with gmpxx, use a GMP wrapper with expression templates to reduce the cost of some computations. Since not all platforms that have GMP also have GMPXX (MSVC in particular), this is a second chance. We could also give it priority over GMPXX if you prefer, it doesn't change much.